gradle-conjure/src/test/groovy/com/palantir/gradle/conjure/ConfigurationCacheTest.groovy

```diff
@@ -0,0 +1,107 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle
+
+import nebula.test.IntegrationTestKitSpec
+
+class ConfigurationCacheTest extends IntegrationTestKitSpec {
+
+    def setup() {
+        definePluginOutsideOfPluginBlock = true
+        keepFiles = true
+    }
+
+    def "classes task runs with configuration cache when applying conjure"() {
+        given:
+        buildFile << """
+            apply plugin: 'com.palantir.conjure'
```

Comment:
I'm not sure these tests test the tasks are CC safe. I think we need to make the various conjure subprojects (eg `whatever-conjure-objects`) and add a conjure yml file for it to actually make the various generator/compilation tasks.

Reply:
It might be better in these cases to just convert the existing tests to IntegrationTestKitSpec like we've done elsewhere